File: Programming Language Structures/File Conversion-- remove control blocking/main-1.py
import sys
def main():
    #Open and create a new file ouput
    with open("control-char.txt", "r") as input_file, open("outcome.txt", "w") as output_file:
        inc = False
        outc = False
        #loop througout the file and check for control c and control b
        for line in input_file:
            length = len(line)

            for i in range(length):
                if ord(line[i]) == 3 and inc==False:
                    inc = True
                    outc = False
                if ord(line[i]) == 2 and inc == True:
                    inc = False
                    outc = True
                if not inc and not outc:

                        output_file.write(line[i])
                if outc:
                    outc = False
        input_file.close()
        output_file.close()

if __name__=="__main__":
    main()

File: Programming Language Structures/File Conversion-- remove control blocking/Main-1.java
package SQL;


import java.io.*;
import java.util.Scanner;

public class Main {
    public static void main(String[] args) {
        try {
            // open files and create a new file
            FileReader input=new FileReader("control-char.txt");
            Scanner s = new Scanner(input);
            FileWriter out= new FileWriter("outcome.txt");
            boolean inc=false;
            boolean outc=false;
            String line;
            // loop throught the file and check for control c and control b
            while (s.hasNextLine()){
                line=s.nextLine();
                int length = line.length();

                for (int i =0 ; i < length;i++){
                    if( (int)line.charAt(i)==3 && inc==false){
                        inc=true;
                        outc=false;
                    }
                    if((int)line.charAt(i)==2 && inc==true){
                        inc=false;
                        outc=true;
                    }
                    if(inc == false && outc==false){

                            out.write(line.charAt(i));

                    }
                    if(outc == true){
                        outc=false;

                    }
                }
                if(inc==false){
                    out.write("\n");
                }

            }
            input.close();
            out.close();
        } catch (IOException e) {
            e.printStackTrace();
        }






    }
}
